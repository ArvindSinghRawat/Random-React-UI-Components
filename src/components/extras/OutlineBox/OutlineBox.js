import React from 'react';

// Style
import styles from './OutlineBox.module.css';

export const OutlineBox = (props) => {
  const cssClasses = [styles.OutlineBox, styles.Rounded];
  return (
    <div
      className={cssClasses.join(' ') + ' ' + props.className}
      style={{ ...props.style }}
    >
      {props.children}
    </div>
  );
};
