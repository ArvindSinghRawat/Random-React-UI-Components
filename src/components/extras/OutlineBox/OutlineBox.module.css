.OutlineBox {
  outline: none;
  appearance: none;
  border: 1px solid #ccc;
  box-shadow: 0 0 1px 1px #eee;
}
.OutlineBox.RoundedXS{
    border-radius: 0.2em;
}
.OutlineBox.RoundedSM{
    border-radius: 0.4em;
}
.OutlineBox.RoundedMD,
.OutlineBox.Rounded{
    border-radius: 0.8em;
}
.OutlineBox.RoundedLG{
    border-radius: 1.6em;
}
.OutlineBox.RoundedXL{
    border-radius: 3.2em;
}
