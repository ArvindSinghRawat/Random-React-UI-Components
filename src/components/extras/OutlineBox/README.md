# Outline Box
## Introduction
This is a wrapper that allows to wrap content with a nice thin border.

#### Image of an Empty Outline Box
![Image of OutlineBox in Action](https://github.com/ArvindSinghRawat/Random-React-UI-Components/blob/master/src/components/extras/OutlineBox/screenshots/EmptyOutlineBox.png?raw=true)

#### Image of Outline Box with Content
![Image of OutlineBox with Content](https://github.com/ArvindSinghRawat/Random-React-UI-Components/blob/master/src/components/extras/OutlineBox/screenshots/OutlineBoxWithContent.png?raw=true)
