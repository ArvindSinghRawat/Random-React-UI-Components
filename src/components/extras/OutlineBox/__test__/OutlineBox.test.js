import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { OutlineBox } from '../OutlineBox';

configure({ adapter: new Adapter() });

describe('<OutlineBox />', () => {
  let wrapper = null;
  const content = 'This is Outline Box.';
  beforeEach(() => {
    wrapper = shallow(<OutlineBox>{content}</OutlineBox>);
  });

  it('Should render <OutlineBox />', () => {
    expect(wrapper.text()).toMatch(content);
  });
});
