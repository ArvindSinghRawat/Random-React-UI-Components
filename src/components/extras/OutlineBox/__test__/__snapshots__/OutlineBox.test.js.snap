// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OutlineBox /> Should render <OutlineBox /> 1`] = `
<div
  className="OutlineBox Rounded"
  style={Object {}}
>
  This is Outline Box.
</div>
`;

exports[`<OutlineBox /> Should render <OutlineBox className="customClass"/> 1`] = `
<div
  className="OutlineBox Rounded customClass"
  style={Object {}}
>
  This is Outline Box.
</div>
`;
